linux/keyman-config/keyman_config/ibus_util.py

```diff
@@ -68,15 +69,21 @@ def restart_ibus_subp():
 
 
 def restart_ibus(bus=None):
-    try:
-        if not bus:
-            bus = get_ibus_bus()
-        logging.info("restarting IBus")
-        bus.exit(True)
-        bus.destroy()
-    except Exception as e:
-        logging.warning("Failed to restart IBus")
-        logging.warning(e)
+    realuser = os.environ.get('SUDO_USER')
+    if realuser:
+        # we have been called with `sudo`. Restart ibus for the real user.
+        logging.info('restarting IBus by subprocess for user %s', realuser)
+        subprocess.run(['sudo', '-u', realuser, 'ibus', 'restart'])
```

Comment:
LGTM but I wonder if it is more coherent to actually call ourselves rather than call out to `ibus`? It just feels a bit funny to be restarting ibus in two different methods, depending on whether or not we are sudo'd.

Reply:
Yes, that would be nicer. Unfortunately, when running as `sudo` we don't have access to the user's ibus interface. That would still leave the option of calling `ibus restart` for the regular user, but I'm hesitant because of potential side effects.

Reply:
Yes, that would be nicer. Unfortunately, when running as `sudo` we don't have access to the user's ibus interface. That would still leave the option of calling `ibus restart` for the regular user, but I'm hesitant because of potential side effects.

Reply:
I would expect us to call into a helper python script to do the restart (e.g. something like `ibus_util.py -restart-ibus`)